From 21af533bf5848e9467c3a0cba622de89d94e1df9 Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Tue, 4 Sep 2018 16:53:21 +0800
Subject: [PATCH 4511/5242] MLK-19444 VPU: Decrease encoder wait time from 1s
 to 10ms and remove vpu_set_power for QXP in probe
 function

commit  2d193c6a6165d1eb4e8508bdb07014262cbd71fa from
https://source.codeaurora.org/external/imx/linux-imx.git

Decrease encoder wait time from 1s to 10ms and remove vpu_set_power for
QXP in probe function

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c |   11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
index 3b8c369..14cae38 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
@@ -875,9 +875,9 @@ static bool update_yuv_addr(struct vpu_ctx *ctx, u_int32 uStrIdx)
 	while (1) {
 		if (!wait_event_interruptible_timeout(ctx->buffer_wq_input,
 				(!list_empty(&This->drv_q)) || ctx->forceStop,
-				msecs_to_jiffies(1000))) {
+				msecs_to_jiffies(10))) {
 			if (!ctx->forceStop)
-				vpu_dbg(LVL_ERR, " warn: yuv wait_event_interruptible_timeout wait 1s timeout\n");
+				vpu_dbg(LVL_INFO, " warn: yuv wait_event_interruptible_timeout wait 10ms timeout\n");
 			else
 				break;
 		}
@@ -959,9 +959,9 @@ static void report_stream_done(struct vpu_ctx *ctx,  MEDIAIP_ENC_PIC_INFO *pEncP
 	while (1) {
 		if (!wait_event_interruptible_timeout(ctx->buffer_wq_output,
 				(!list_empty(&This->drv_q)),
-				msecs_to_jiffies(1000))) {
+				msecs_to_jiffies(10))) {
 			if (!ctx->forceStop)
-				vpu_dbg(LVL_ERR, " warn: stream wait_event_interruptible_timeout wait 1s timeout\n");
+				vpu_dbg(LVL_INFO, " warn: stream wait_event_interruptible_timeout wait 10ms timeout\n");
 			else
 				break;
 		}
@@ -2160,8 +2160,7 @@ static int vpu_probe(struct platform_device *pdev)
 	if (core_type == 2) {
 		vpu_set_power(dev, true, 0);
 		vpu_set_power(dev, true, 1);
-	} else
-		vpu_set_power(dev, true, 0);
+	}
 
 	vpu_enable_hw(dev);
 
-- 
1.7.9.5

